test/spec/proxyBenchmark.js

```diff
@@ -58,69 +43,153 @@ function generateBigObjectFixture(carsSize) {
-  obj.phoneNumbers[1].number = '456';
-  jsonpatch.generate(observer);
-});
-
-suite.add('jsonpatcherproxy mutation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.cars[50].name = 'Toyota'
-});
-
-suite.add('jsonpatch mutation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const observer = jsonpatch.observe(obj);
-  obj.cars[50].name = 'Toyota';
-  jsonpatch.generate(observer);
-});
-
-suite.add('jsonpatcherproxy generate operation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.cars.shift();
-});
-
-suite.add('jsonpatch generate operation - huge object', function() {
-  const obj = generateBigObjectFixture(100);
-  const observer = jsonpatch.observe(obj);
-  obj.cars.shift();
-  jsonpatch.generate(observer);
-});
-
-suite.add('PROXIFY big object', function() {
-  const obj = generateBigObjectFixture(50);
-  const jsonPatcherProxy = new JSONPatcherProxy(obj);
-  const observedObj = jsonPatcherProxy.observe(true);
-  observedObj.a = 1;
-});
-
-suite.add('DIRTY-OBSERVE big object', function() {
-  const obj = generateBigObjectFixture(50);
-  const observer = jsonpatch.observe(obj);
-  obj.a = 1;
-  jsonpatch.generate(observer);
-
-});
+  obj.cars[0].speed = 123;
+  obj.cars[0].temp.speed = 456;
+}
+
+function reverseString(str) {
+  return str.split("").reverse().join("");
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Observe and generate, small object';
+  
+  {
+    suite.add(`${suiteName} (noop)`, function() {
+      const obj = generateBigObjectFixture(1);
+      modifyObj(obj);
+    });
+  }
+
+  {
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      const obj = generateBigObjectFixture(1);
+      const jsonPatcherProxy = new JSONPatcherProxy(obj);
+      const observedObj = jsonPatcherProxy.observe(true);
+      const patches = jsonPatcherProxy.generate();
+      modifyObj(observedObj);
+    });
+  }
+  
+  {
+    suite.add(`${suiteName} (fast-json-patch)`, function() {
+      const obj = generateBigObjectFixture(1);
+      const observer = jsonpatch.observe(obj);
+      modifyObj(obj);
+      jsonpatch.generate(observer);
+    });
+  }
+}
+
+/* ============================= */
+
+{
+  const suiteName = 'Observe and generate';
+  
+  {
+    suite.add(`${suiteName} (noop)`, function() {
+      const obj = generateBigObjectFixture(100);
+      modifyObj(obj);
+    });
+  }
+
+  {
+    suite.add(`${suiteName} (JSONPatcherProxy)`, function() {
+      const obj = generateBigObjectFixture(100);
+      const jsonPatcherProxy = new JSONPatcherProxy(obj);
+      const observedObj = jsonPatcherProxy.observe(true);
+      const patches = jsonPatcherProxy.generate();
```

Comment:
I think you should generate patches after you make changes,

Reply:
I didn't change this. It is how it works on `master`. But I think you are right, I will change it. 

@alshakero do you remember why did you chose to call `generate` here not later?

Reply:
I think line 106 can be removed altogether. See https://github.com/Palindrom/JSONPatcherProxy/pull/39/files/bcde61c41f9ebe5487fe4a5c712bb2d911a064e9..67cc6f570e275fcb4e1c9a886eb9977e3421bf83#r291562016

Reply:
Ah that would be a leak as shown in @warpech comment https://github.com/Palindrom/JSONPatcherProxy/pull/39#issuecomment-499857818